### PR TITLE
Added support for Array type

### PR DESCRIPTION
**Pull request by Hajto**
- Added support for Arrays
 - Added tests for Seaport protocol eip712
 - Fixed bugs with encoding.

### CODE REVIEW

File: lib/ex_web3_ec_recover/signed_type.ex [hunk 1]
```diff
@@ -31,27 +31,45 @@ defmodule ExWeb3EcRecover.SignedType do
   """
   @spec hash_message(map(), types(), String.t(), Keyword.t()) :: hash :: binary()
   def hash_message(message, types, primary_type, opts \\ []) do
+    # TODO: Remove
```
Comment (grahac) [inline lib/ex_web3_ec_recover/signed_type.ex]: When should we remove this?

File: test/real_world_cases/seaport_protocol_test.exs [hunk 2]
```diff
@@ -0,0 +1,86 @@
+defmodule ExWeb3EcRecover.RealWorldCases.SeaportProtocolTest do
```
Comment (grahac) [inline test/real_world_cases/seaport_protocol_test.exs:1]: Can we change the name here to full payload test?

**Review by grahac — CHANGES_REQUESTED** [review]

File: test/real_world_cases/seaport_protocol_test.exs (hunk 2 above)
Comment (Hajto) [reply test/real_world_cases/seaport_protocol_test.exs:1]: Intention of this group of modules is to test real world cases. Like 0x (previous real world case) and now Seaport. It tests the lib working on real world test cases. It literally encodes a Seaport OrderComponents.

File: lib/ex_web3_ec_recover/signed_type.ex (hunk 1 above)
Comment (Hajto) [reply lib/ex_web3_ec_recover/signed_type.ex]: Now. Forgot to remove the message.

File: test/real_world_cases/seaport_protocol_test.exs (hunk 2 above)
Comment (grahac) [reply test/real_world_cases/seaport_protocol_test.exs:1]: ok [](edited-with-Axolo)

**Review by emoragaf — APPROVED** [review]

**Review by ZuraGuerra — APPROVED** [review]

**Pull request merged**